test/openAPI/features/wallet_binding.feature

```diff
@@ -0,0 +1,31 @@
+@method=POST @endpoint=/wallet-binding
+Feature: The endpoint to validate wallet create wallet user id
+
+  @smoke @unit @positive
+  Scenario: Successfully validates the wallet and generates the wallet user id smoke type test
+    Given Wants to validate the wallet and generate wallet user id
+    When Send POST /wallet-binding request with given "n3fy2qkg9r7h2" as individualId and "OTP" as authFactorType and "encoded-json" as format and "string" as challenge and publicKey
+    Then Receive a response from the /wallet-binding endpoint
+    And The /wallet-binding response should be returned in a timely manner 15000ms
+    And The /wallet-binding endpoint response should have status 200
+    And The /wallet-binding endpoint response should match json schema with no errors
+
+  @negative
+  Scenario: Not able to generate the wallet binding because of unsupported challenge format
+    Given Wants to validate the wallet and generate wallet user id
+    When Send POST /wallet-binding request with given invalid format "n3fy2qkg9r7h2" as individualId and "OTP" as authFactorType and "invalid_format" as format and "string" as challenge and publicKey
+    Then Receive a response from the /wallet-binding endpoint
+    And The /wallet-binding response should be returned in a timely manner 15000ms
+    And The /wallet-binding endpoint response should have status 200
+    And The /wallet-binding endpoint response should match json schema with errors
+    And The /wallet-binding response should contain errorCode property equals to "unsupported_challenge_format"
+
+  @negative
+  Scenario: Not able to generate the wallet binding because of invalid public key
+    Given Wants to validate the wallet and generate wallet user id
+    When Send POST /wallet-binding request with given format "n3fy2qkg9r7h2" as individualId and "OTP" as authFactorType and "encoded-json" as format and "string" as challenge and invalid publicKey
+    Then Receive a response from the /wallet-binding endpoint
+    And The /wallet-binding response should be returned in a timely manner 15000ms
+    And The /wallet-binding endpoint response should have status 200
+    And The /wallet-binding endpoint response should match json schema with errors
+    And The /wallet-binding response should contain errorCode property equals to "invalid_public_key"
```

Comment:
Can we add one more test scenario with calling the wallet binding twice with same public key and expect "duplicate_public_key" as error code ?

Reply:
Tried to test the duplicate public key, but it was failing instantly with an error: `Duplicate request initiated` and could compare if the public_key is duplicated. I think now we have to keep it that way and later on we might be able to compare public key with software

Reply:
@vishwa-vyom updated changed you can find there -> https://github.com/GovStackWorkingGroup/bb-identity/pull/43

Reply:
In the #43 I am not able to find any scenario added for this

Reply:
I added scenario for "duplicate_public_key" ;)

Reply:
Actually I am not following the implementation of this test case, we have to send wallet binding request twice with same public key to achieve this.

Reply:
And this is what I did but I see now it's not so obvious because of not clear scenario :D 
I will fix it, thanks!